core/store/migrate/migrations/0162_logpoller_block_timestamps.sql

```diff
@@ -0,0 +1,7 @@
+-- +goose Up
+ALTER TABLE evm_logs ADD COLUMN block_timestamp timestamptz NOT NULL DEFAULT now();
```

Comment:
The DEFAULT Is  only there for the migration.  You must have a DEFAULT set in order to add a column to a non-empty table.
Only other options besides DEFAULT now() I can think of are:

```
1.   purge all logs from the table during migration
or
2.  use NULL default
or
3.  use some very old date, like 1970, for the default.
```

I assume the first one is off the table.  The problem with the second one is that you can't read the `block_timestamp` column directly into a `logpoller.Log` struct unless you define it as a pointer and allow it to be nil... otherwise the datatypes don't match.  I decided that would make things uglier in several ways, and means you've got to add permanent checks for nil in lots of places just to solve this one-time migration issue.

I decided it was cleaner and simpler to just define it as `time.Time` instead of `*time.Time`:
```
type Log struct {
	EvmChainId     *utils.Big
	LogIndex       int64
	BlockHash      common.Hash
	BlockNumber    int64
	BlockTimestamp time.Time
	Topics         pq.ByteaArray
	EventSig       common.Hash
	Address        common.Address
	TxHash         common.Hash
	Data           []byte
	CreatedAt      time.Time
}
```
This seems cleaner and simpler in several ways.  It would be confusing  and possibly error-prone  if `CreatedAt` was `time.Time` but `BlockTimestamp` was `*time.Time`.

The 3rd option I'm definitely open to, if that would make more sense... although I don't think I have enough context on the use cases for this feature to know which would be better.  Any preference there?  Also open to other ideas not on my list, if I've missed one.

Reply:
Oh, I suppose a slightly better variant of this might be:  whether we choose now() or [old date] for the DEFAULT during migration, we could add a second step after the ALTER COLUMN which drops the default.  I think I like that approach actually... fits better with not wanting a permanent solution for a temporary problem.  Either way though, could use advice on what the best timestamp is to fill in for all pre-migration logs.

Reply:
Could we set it to the existing createdAt? That would be roughly accurate

Reply:
Yeah, I don't know of any way we could access that in the migration scripts.

Reply:
If it was really important to have these be accurate, we do have one option but it seems like overkill to me:  set them all to 1970, then during node startup have a second phase of migration where it re-requests all blocks set to a 1970 timestamp and updates them to accurate timestamp.
I don't know of a way it could skip that after the first restart though... it would probably at least have to run the query to find if any of them are set to 1970 each time it starts up.

Assuming this gets rolled out in the next release (or any release <= CCIP release), then it doesn't seem like something we should care much, right?  If CCIP is the only customer and that won't be enabled until later... won't these blocks be far in the past by then?

Reply:
Correct shouldn't be an issue because of the CCIP release sequencing. I think that + the fact that we can always delete and replay manually to repopulate the correct data as a recovery technique makes DEFAULT now() fine.